holmes/core/supabase_dal.py

```diff
@@ -262,5 +266,64 @@ def get_workload_issues(self, resource: dict, since_hours: float) -> List[str]:
             return data
 
         except:
-            logging.exception("failed to fetch workload issues data")
+            logging.exception("failed to fetch workload issues data", exc_info=True)
+            return []
+
+    def upsert_holmes_status(self, holmes_status_data: dict) -> None:
+        updated_at = datetime.now().isoformat()
+        try:
+            res = (
+                self.client
+                .table(HOLMES_STATUS_TABLE)
+                .upsert({
+                    "account_id": self.account_id,
+                    "updated_at": updated_at,
+                    **holmes_status_data,
+                },
+                on_conflict='account_id, cluster_id')
+                .execute()
+            )
+        except Exception as error:
+            logging.error(f"Error happened during upserting holmes status: {error}", 
+                          exc_info=True)
+
+        return None
+    
+    def sync_toolsets(self, toolsets: list[dict]) -> None:
+        if not toolsets:
+            logging.warning("No toolsets were provided for synchronization.")
+            return
+        
+        provided_toolset_names = [toolset['toolset_name'] for toolset in toolsets]
+   
+        try:
+            self.client.table(HOLMES_TOOLSET).upsert(
+                toolsets,
+                on_conflict='account_id, cluster_id, toolset_name'
+            ).execute()
+
+            logging.info("Toolsets upserted successfully.")
+
+            
+            self.client.table(HOLMES_TOOLSET).delete().eq("account_id", 
+                                                                       self.account_id).eq(
+                                                                           'cluster_id', CLUSTER_NAME).not_.in_(
+                'toolset_name', provided_toolset_names
+            ).execute()
+
+            logging.info("Toolsets synchronized successfully.")
+
+        except Exception as e:
+            logging.exception(f"An error occurred during toolset synchronization: {e}",
+                              exc_info=True)
+
+    def get_toolsets_for_holmes(self,):
```

Comment:
why are we getting this from the db?

The enabled/disabled status is coming from helm
We're saving it to report the status to the saas, but to read it, we have it locally (locally is the source of truth)